cmd/multiplexer/run.go

```diff
@@ -76,15 +104,61 @@ func runAction(c *cli.Context) error {
 		errChan <- mux.Run(ctx)
 	}()
 
+	// Start grpc server.
+	grpcServer := grpc.NewServer()
+	reflection.Register(grpcServer)
+
+	server, err := newServer(creator, registry)
+	if err != nil {
+		return err
+	}
+
+	lnmux_proto.RegisterMultiplexerServer(
+		grpcServer, server,
+	)
+
+	grpcInternalListener, err := net.Listen("tcp", ":19090")
+	if err != nil {
+		return err
+	}
+
+	grpcErrChan := make(chan error)
+	go func() {
+		log.Infow("Grpc server startin on port 19090")
+		err := grpcServer.Serve(grpcInternalListener)
+		if err != nil && err != grpc.ErrServerStopped {
+			grpcErrChan <- fmt.Errorf("grpc server error: %v", err)
+			return
+		}
+
+		log.Infow("Internal grpc server stopped")
+		grpcErrChan <- nil
+	}()
+
 	// Wait for break and terminate.
 	log.Infof("Press ctrl-c to exit")
 	sigint := make(chan os.Signal, 1)
 	signal.Notify(sigint, os.Interrupt, syscall.SIGINT, syscall.SIGTERM)
 	<-sigint
-	log.Infof("Exiting")
 
+	// Stop grpc server.
+	log.Infof("Stopping grpc server")
+	grpcServer.Stop()
+	err = <-grpcErrChan
+	if err != nil {
+		return err
+	}
+
+	// Stop multiplexer.
+	log.Infof("Stopping multiplexer")
 	cancel()
-	return <-errChan
+	err = <-errChan
```

Comment:
is there's an error returned by returned `mux.Run()` we won't be able to catch it here until `sigint` channel returns a msg. Perhaps you should select over `sigint` and `errChan` chans

Reply:
Fixed